tests/tornado/test_hyperbahn.py

```diff
@@ -44,6 +46,31 @@ def test_new_client_establishes_peers():
         assert channel.peers.lookup(router)
 
 
+def test_new_client_establishes_peers_from_file():
+
+    host_path = os.path.join(
+        os.path.dirname(os.path.realpath(__file__)),
+        '../data/hosts.json',
+    )
+
+    # TChannel knows about one of the peers already.
+    channel = TChannel('test', known_peers=['127.0.0.1:23002'])
+
+    hyperbahn.advertise(
+        channel,
+        'baz',
+        None,
+        None,
+        host_path
+    )
```

Comment:
@abhinav, 1) I meant to test hyperbahn.advertise. 2) Will using yield make any difference in this test.